### PR TITLE
feat(storefront): Refactor storefinder component to CmsComponent with child routes

### PR DESCRIPTION
**Pull request by dunqan**
Closes GH-1655

### CODE REVIEW

File: projects/storefrontlib/src/lib/store-finder/store-finder.module.ts [hunk 1]
```diff
@@ -32,7 +39,41 @@ import { StoreFinderCoreModule, UrlTranslationModule } from '@spartacus/core';
...
+            {
+              path: 'find',
+              component: StoreFinderSearchResultComponent
+            },
+            {
+              path: 'view-all',
+              component: StoreFinderStoresCountComponent
+            },
+            {
+              path: 'country/:country' /* 'country/:country/region/:region', */,
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/store-finder/store-finder.module.ts]: Why there is the comment `/* 'country/:country/region/:region', */`?

File: projects/storefrontlib/src/lib/ui/pages/store-finder-page/store-finder-page.module.ts [hunk 2]
```diff
@@ -1,66 +0,0 @@
...
-import { StoreFinderPageLayoutModule } from '../../layout/store-finder-page-layout/store-finder-page-layout.module';
-// tslint:disable-next-line:max-line-length
-import { StoreFinderSearchResultComponent } from '../../../store-finder/components/store-finder-search-result/store-finder-search-result.component';
-// tslint:disable-next-line:max-line-length
-import { StoreFinderStoresCountComponent } from '../../../store-finder/components/store-finder-stores-count/store-finder-stores-count.component';
-import { StoreFinderGridComponent } from '../../../store-finder/components/store-finder-grid/store-finder-grid.component';
-// tslint:disable-next-line:max-line-length
-import { StoreFinderStoreDescriptionComponent } from '../../../store-finder/components/store-finder-store-description/store-finder-store-description.component';
-
-const routes: Routes = [
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/ui/pages/store-finder-page/store-finder-page.module.ts]: Now we can also remove `storeFinder` from `default-storefront-routes-translations.ts`

Comment (dunqan) [reply projects/storefrontlib/src/lib/ui/pages/store-finder-page/store-finder-page.module.ts]: 👍

File: projects/storefrontlib/src/lib/store-finder/store-finder.module.ts (hunk 1 above)
Comment (dunqan) [reply projects/storefrontlib/src/lib/store-finder/store-finder.module.ts]: Thanks! Left comment to fix it later, but missed it somehow ;-)

**Pull request merged**